scala/ppml/src/main/java/com/intel/analytics/bigdl/ppml/psi/PsiIntersection.java

```diff
@@ -131,7 +131,7 @@ public FindIntersection(String[] a,
 
     public List<String> getIntersection() throws InterruptedException{
         synchronized (this) {
-            if(null == intersection) {
+            while (intersection == null) {
```

Comment:
timeout or wait indefinitely?

Reply:
Previous code waits indefinitely, I think could change to:
* server does not block, just return
* client have a timeout and retry mechanism, and have a max_retry